javalib/src/main/scala/java/util/TreeSet.scala

```diff
@@ -0,0 +1,222 @@
+        } else {
+          last.foreach(self.remove(_))
+          last = None
+        }
+      }
+    }
+  }
+
+  def descendingSet(): NavigableSet[E] = {
+    val descSetFun = { () =>
+      val retSet = new mutable.TreeSet[Box[E]]()(BoxOrdering.reverse)
+      retSet.addAll(inner)
+      retSet
+    }
+    new NavigableView(this, descSetFun)
+  }
+
+  def size(): Int =
+    inner.size
+
+  override def isEmpty(): Boolean =
+    inner.isEmpty
+
+  override def contains(o: Any): Boolean =
+    if (o == null) false
+    else inner.contains(Box(o.asInstanceOf[E]))
+
+  override def add(e: E): Boolean =
+    if (e == null) throw new NullPointerException()
+    else inner.add(Box(e))
+
+  override def remove(o: Any): Boolean =
+    if (o == null) throw new NullPointerException()
+    else inner.remove(Box(o.asInstanceOf[E]))
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = add(iter.next()) || changed
+    changed
+  }
+
+  override def removeAll(c: Collection[_]): Boolean = {
+    val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = inner.remove(Box(iter.next).asInstanceOf[Box[E]]) || changed
+    changed
+  }
+
+  def subSet(fromElement: E, fromInclusive: Boolean, toElement: E, toInclusive: Boolean): NavigableSet[E] = {
+    val subSetFun = { () =>
+      val boxedFrom = Box(fromElement)
+      val boxedTo = Box(toElement)
+      val ss = new mutable.TreeSet[Box[E]]()
+      ss ++= inner.range(boxedFrom, boxedTo)
+      if (!fromInclusive && inner.contains(boxedFrom))
+        ss.remove(boxedFrom)
+      if (toInclusive && inner.contains(boxedTo))
+        ss += boxedTo
+      ss
+    }
+    new NavigableView(this, subSetFun)
+  }
+
+  def headSet(toElement: E, inclusive: Boolean): NavigableSet[E] = {
+    val headSetFun = { () =>
+      val boxed = Box(toElement)
+      val hs = new mutable.TreeSet[Box[E]]()
+      hs ++= inner.until(boxed)
+      if (inclusive && inner.contains(boxed))
+        hs += boxed
+      hs
+    }
+    new NavigableView(this, headSetFun)
+  }
+
+  def tailSet(fromElement: E, inclusive: Boolean): NavigableSet[E] =  {
+    val tailSetFun = { () =>
+      val boxed = Box(fromElement)
+      val ts = new mutable.TreeSet[Box[E]]()
+      ts ++= inner.from(boxed)
+      if (!inclusive && inner.contains(boxed))
+        ts.remove(boxed)
+      ts
+    }
+    new NavigableView(this, tailSetFun)
+  }
+
+  def subSet(fromElement: E, toElement: E): NavigableSet[E] =
+    subSet(fromElement, true, toElement, false)
+
+  def headSet(toElement: E): NavigableSet[E] =
+    headSet(toElement, false)
+
+  def tailSet(fromElement: E): NavigableSet[E] =
+    tailSet(fromElement, true)
+
+  def comparator(): Comparator[_ >: E] = _comparator
+
+  def first(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
```

Comment:
You don't need to check and throw the exception, `inner.head.inner` will do it for you.